grammars/puppet.cson

```diff
@@ -7,166 +7,32 @@
 'name': 'Puppet'
 'patterns': [
   {
-    'include': '#line_comment'
+    'include': '#puppet'
   }
   {
-    'include': '#constants'
+    'include': '#definitions'
   }
-  {
-    'begin': '^\\s*/\\*'
-    'end': '\\*/'
-    'name': 'comment.block.puppet'
-  }
-  {
-    'begin': '^\\s*(node|class)\\s+((?:[-_A-Za-z0-9"\'.]+::)*[-_A-Za-z0-9"\'.]+)\\s*'
-    'captures':
-      '1':
-        'name': 'storage.type.puppet'
-      '2':
-        'name': 'entity.name.type.class.puppet'
-    'end': '(?={)'
-    'name': 'meta.definition.class.puppet'
-    'patterns': [
-      {
-        'begin': '\\b(inherits)\\b\\s+'
-        'captures':
-          '1':
-            'name': 'storage.modifier.puppet'
-        'end': '(?={)'
-        'name': 'meta.definition.class.inherits.puppet'
-        'patterns': [
-          {
-            'match': '\\b((?:[-_A-Za-z0-9".]+::)*[-_A-Za-z0-9".]+)\\b'
-            'name': 'support.type.puppet'
-          }
-        ]
-      }
-      {
-        'captures':
-          '1':
-            'name': 'variable.other.puppet'
-          '2':
-            'name': 'punctuation.definition.variable.puppet'
-        'match': '((\\$+)[a-zA-Z_][a-zA-Z0-9_]*)\\s*(?=,|\\))'
-        'name': 'meta.function.argument.no-default.puppet'
-      }
-      {
-        'begin': '((\\$+)[a-zA-Z_][a-zA-Z0-9_]*)(?:\\s*(=)\\s*)\\s*'
-        'captures':
-          '1':
-            'name': 'variable.other.puppet'
-          '2':
-            'name': 'punctuation.definition.variable.puppet'
-          '3':
-            'name': 'keyword.operator.assignment.puppet'
-        'end': '(?=,|\\))'
-        'name': 'meta.function.argument.default.puppet'
-        'patterns': [
-          {
-            'include': '#parameter-default-types'
-          }
-        ]
-      }
-    ]
-  }
-  {
-    'begin': '^\\s*(define)\\s+([a-zA-Z0-9_:]+)\\s*(\\()'
+]
+'repository':
+  'functions':
+    'begin': '(?:^|\\s|\\W|\\D)(\\w+)\\s*(\\()'
     'beginCaptures':
       '1':
-        'name': 'storage.type.function.puppet'
+        'name': 'support.function.puppet'
       '2':
-        'name': 'entity.name.function.puppet'
-      '3':
-        'name': 'punctuation.definition.parameters.begin.puppet'
-    'contentName': 'meta.function.arguments.puppet'
+        'name': 'punctuation.definition.function.begin.puppet'
```

Comment:
`[...]function.begin.bracket.round.puppet`

^ That's the new style being introduced in a couple of languages, so it'd be awesome if you could follow that as well.  Same below with the end capture, and everywhere else.